include/dxc/Support/D3DReflection.h

```diff
@@ -0,0 +1,25 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxReflection.h                                                            //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Provides the needed headers and defines for D3D reflection.               //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#ifndef _WIN32
+// need to disable this as it is voilated by this header
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnon-virtual-dtor"
```

Comment:
@pow2clk Didn't we disable this warning altogether as discussed in https://github.com/microsoft/DirectXShaderCompiler/pull/3793#pullrequestreview-861757581, or is it now enabled by default on newer compilers (since I only commented out, instead of leaving an unconditional `-Wnon-virtual-dtor`)?